### PR TITLE
#1169: fix id check bug; defensive programming in context menu update/create logic

### LINKED ISSUE #1169: Duplicate context menu extension entries
- In context menu
- In the page editor (appear to all have the same extension id)

### PR DESCRIPTION
**Pull request by twschiller**
Closes #1169

### CODE REVIEW

File: src/background/contextMenus.ts [hunk 1]
```diff
@@ -152,53 +163,46 @@ export const ensureContextMenu = liftBackground(
...
-        console.debug(`Created new context menu: ${extensionId}`, {
-          menuId,
-          title,
-          contexts,
-          documentUrlPatterns,
-          extensionId,
-        });
+        // Just to be safe if our `extensionMenuItems` book keeping is off, remove any stale menu item
+        try {
+          await browser.contextMenus.remove(expectedMenuId);
```
Comment (fregante) [inline src/background/contextMenus.ts:181]: In these cases I've been just appending a catch with a noop, it looks less noisy than an empty try/catch

File: src/background/contextMenus.ts [hunk 2]
```diff
@@ -152,53 +163,46 @@ export const ensureContextMenu = liftBackground(
...
+        createdMenuId = browser.contextMenus.create(
+          {
+            ...createProperties,
+            id: makeMenuId(extensionId),
+          },
+          () => {
+            resolve(createdMenuId);
+          }
+        );
+      });
```
Comment (fregante) [inline src/background/contextMenus.ts:203]: A bit roundabout. The types are unfortunately right and the API is very awkward to use. I suggest:

```JavaScript 
const {promise, resolve} = pDefer())
const menuId = browser.contextMenus.create(etc, resolve)
await promise;
```

However the callback might still trigger `chrome.runtime.lastError` so you might want to check that in the callback, like:

``` JavaScript 

const menuId = browser.contextMenus.create(etc, () => {
  if (chrome.runtime.lastError) {
    reject(new Error(chrome.runtime.lastError.message))
} else resolve()
})
await promise;

Comment (twschiller) [reply src/background/contextMenus.ts:216]: You beat me to it - yep I've added that check

File: src/options/slices.ts [hunk 3]
```diff
@@ -298,7 +298,7 @@ export const optionsSlice = createSlice({
       const persistedId = extensionId ?? id;
 
       // Support both extensionId and id to keep the API consistent with the shape of the stored extension
-      if (persistedId) {
+      if (persistedId == null) {
```
Comment (twschiller) [inline src/options/slices.ts:301]: This is the bug I introduced in the previous PR

File: src/background/contextMenus.ts (hunk 2 above)
Comment (fregante) [reply src/background/contextMenus.ts:216]: Still roundabout solution though, two variables that are identical, one set synchronously and the other one set via promise

**Pull request merged**